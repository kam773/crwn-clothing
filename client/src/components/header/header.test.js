import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { Header } from './header.component';
import { MemoryRouter } from 'react-router-dom';
import { OptionsContainer } from './header.styles';

const mockProps = {
  currentUser: {
    displayName: 'name',
    id: 'asfssdg12',
    email: 'email@example.com',
  },
  hidden: true,
};

describe('create snapshot', () => {
  it('captures snapshot of Header', () => {
    const wrapper = renderer
      .create(
        <MemoryRouter>
          <Header {...mockProps} />
        </MemoryRouter>,
      )
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});

describe('shallow render react component', () => {
  it('renders DUMB react component', () => {
    const wrapper = shallow(<Header {...mockProps} />);
    expect(wrapper.length).toEqual(1);
  });
  it('renders Logo component', () => {
    const wrapper = shallow(<Header {...mockProps} />);
    expect(wrapper.find('Logo')).toBeDefined();
  });
  it('renders 4 nav items', () => {
    const wrapper = shallow(<Header {...mockProps} />);
    expect(wrapper.find(OptionsContainer).children()).toHaveLength(4);
  });
  it('renders CartDropdown component', () => {
    const wrapper = shallow(<Header {...mockProps} />);
    expect(wrapper.find('CartDropdown')).toBeDefined();
  });
});
