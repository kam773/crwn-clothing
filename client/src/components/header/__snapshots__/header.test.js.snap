// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create snapshot captures snapshot of Header 1`] = `
<div
  className="sc-gZMcBi dKLNdS"
>
  <a
    className="sc-gqjmRU BQCF"
    href="/"
    onClick={[Function]}
  >
    <svg
      className="logo"
    >
      crown.svg
    </svg>
  </a>
  <div
    className="sc-VigVT fTbjAp"
  >
    <a
      className="sc-jTzLTM cPNuvf"
      href="/shop"
      onClick={[Function]}
    >
      SHOP
    </a>
    <a
      className="sc-jTzLTM cPNuvf"
      href="/contact"
      onClick={[Function]}
    >
      CONTACT
    </a>
    <div
      className="sc-jTzLTM cPNuvf"
      onClick={[Function]}
    >
      SIGN OUT
    </div>
    <div
      className="sc-bdVaJa kUrIC"
    >
      <svg
        className="sc-bwzfXH cKSSZR"
      >
        shopping-bag.svg
      </svg>
      <span
        className="sc-htpNat fFVhJp"
      />
    </div>
  </div>
</div>
`;
